### DISCUSSION TITLE
Autosave Files

### DISCUSSION THREAD

**Discussion by joncode21** [phcode-dev/phoenix#1776] [Q&A] [answered]
It would be good if the IDE helps us to auto save things rather than asking us if we want to save things.

There might be a situation when  the  computer can suddenly crash and if the file is not saved.It would not be a good  day for the programmer.

**Comment by abose** [comment, marked as answer]
Thanks for the feature request!

* Phoenix code has crash recovery, which should recover unsaved files when you reopen the app.  ![image](https://github.com/user-attachments/assets/a9828a4b-35d2-45b3-9f9d-8998b1d7b495)

## Autosave extension
If you want continuous save, then `Autosave every edit!` extension should help. 
1. open menu and select `File>Extension Manager...`
2. Search for `Autosave every edit!` ![image](https://github.com/user-attachments/assets/7bd2bb16-c810-4f47-8c62-2bd6f3e85da0)
* Then enable the feature using the menu ![image](https://github.com/user-attachments/assets/1650ed03-92c3-437e-a41c-fbdf5b03a0cc)

**Reply by abose** [reply]
@joncode21 Does this solve your need?

**Reply by joncode21** [reply]
YES IT DOES!
